MiniApp/Progress+UIViewController.swift

```diff
@@ -0,0 +1,36 @@
+import UIKit
+
+extension UIViewController {
+
+    func showProgressIndicator() {
+        DispatchQueue.main.async {
+            let alert = UIAlertController(title: nil, message: "Please wait...", preferredStyle: .alert)
+            let loadingIndicator = UIActivityIndicatorView(frame: CGRect(x: 10, y: 5, width: 50, height: 50))
+            loadingIndicator.hidesWhenStopped = true
+            loadingIndicator.style = UIActivityIndicatorView.Style.gray
+            loadingIndicator.startAnimating()
+            alert.view.addSubview(loadingIndicator)
+            self.present(alert, animated: true, completion: nil)
+        }
+    }
+
+    func dismissProgressIndicator() {
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1, execute: {
+            self.dismiss(animated: true, completion: nil)
+        })
+    }
+
+    func displayErrorAlert(title: String, message: String, dimissController: Bool) {
+        DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
+            self.dismiss(animated: true, completion: {
```

Comment:
self.show following self.dismiss, does that work? and `dimissController` param is not used

Reply:
dimissController is used in DisplayController. When the user taps on any mini app from the list (Home screen), we are showing the Displaycontroller and mini app is being fetched in the background. If the downloading fails, we are displaying alert (along with dismissController option) that will dismiss the Display controller and takes the user to the home screen

Reply:
In that case - ok. (Sorry I didn't notice the reference to `dismissController` in the `if` below)